### PR TITLE
559 missing support for validity start date from vc data model v20

### LINKED ISSUE #559: Missing support for validity start date from vc data-model-v2.0
**Describe the bug**
The verifier-api `NotBeforeDatePolicy` doesn't cover the cases for credentials with `validFrom` property, which, according to the [spec](https://www.w3.org/TR/vc-data-model-2.0/#validity-period), is the new name for `issuanceDate`.

**To Reproduce**
1. Go to [issuer-api](https://issuer.portal.walt-test.cloud/swagger/index.html#/Credential%20Issuance/post_openid4vc_jwt_issue)
2. Generate credential-offer (using vc-sample.json from below)
3. Claim credential with [wallet-api](https://wallet.walt-test.cloud/)
4. Go to [verifier-api](https://verifier.portal.walt-test.cloud/swagger/index.html#/Credential%20Verification/post_openid4vc_verify)
5. Generate presentation-request (using vp-sample.json from below)
6. Perform presentation with wallet-api

**Expected behavior**
- popup shows presentation failed
- status code 400

**Actual behavior**
- popup shows presentation successful
- status code 200

**Environment**
- dev: https://verifier.portal.walt-test.cloud
- prod: https://verifier.portal.walt.id
- local

**Additional context**
vc-sample.json
```
{
    "issuerKey":
    {
        "type": "jwk",
        "jwk":
        {
            "kty": "OKP",
            "d": "mDhpwaH6JYSrD2Bq7Cs-pzmsjlLj4EOhxyI-9DM1mFI",
            "crv": "Ed25519",
            "kid": "Vzx7l5fh56F3Pf9aR3DECU5BwfrY6ZJe05aiWYWzan8",
            "x": "T3T4-u1Xz3vAV2JwPNxWfs4pik_JLiArz_WTCvrCFUM"
        }
    },
    "issuerDid": "did:key:z6MkjoRhq1jSNJdLiruSXrFFxagqrztZaXHqHGUTKJbcNywp",
    "credentialConfigurationId": "OpenBadgeCredential_jwt_vc_json",
    "credentialData":
    {
        "@context":
        [
            "https://www.w3.org/2018/credentials/v1",
            "https://purl.imsglobal.org/spec/ob/v3p0/context.json"
        ],
        "id": "urn:uuid:THIS WILL BE REPLACED WITH DYNAMIC DATA FUNCTION (see below)",
        "type":
        [
            "VerifiableCredential",
            "OpenBadgeCredential"
        ],
        "name": "JFF x vc-edu PlugFest 3 Interoperability",
        "issuer":
        {
            "type":
            [
                "Profile"
            ],
            "id": "did:key:THIS WILL BE REPLACED WITH DYNAMIC DATA FUNCTION FROM CONTEXT (see below)",
            "name": "Jobs for the Future (JFF)",
            "url": "https://www.jff.org/",
            "image": "https://w3c-ccg.github.io/vc-ed/plugfest-1-2022/images/JFF_LogoLockup.png"
        },
        "validFrom": "2025-07-20T07:05:44Z",
        "credentialSubject":
        {
            "id": "did:key:123 (THIS WILL BE REPLACED BY DYNAMIC DATA FUNCTION (see below))",
            "type":
            [
                "AchievementSubject"
            ],
            "achievement":
            {
                "id": "urn:uuid:ac254bd5-8fad-4bb1-9d29-efd938536926",
                "type":
                [
                    "Achievement"
                ],
                "name": "JFF x vc-edu PlugFest 3 Interoperability",
                "description": "This wallet supports the use of W3C Verifiable Credentials and has demonstrated interoperability during the presentation request workflow during JFF x VC-EDU PlugFest 3.",
                "criteria":
                {
                    "type": "Criteria",
                    "narrative": "Wallet solutions providers earned this badge by demonstrating interoperability during the presentation request workflow. This includes successfully receiving a presentation request, allowing the holder to select at least two types of verifiable credentials to create a verifiable presentation, returning the presentation to the requestor, and passing verification of the presentation and the included credentials."
                },
                "image":
                {
                    "id": "https://w3c-ccg.github.io/vc-ed/plugfest-3-2023/images/JFF-VC-EDU-PLUGFEST3-badge-image.png",
                    "type": "Image"
                }
            }
        }
    },
    "mapping":
    {
        "id": "<uuid>",
        "issuer":
        {
            "id": "<issuerDid>"
        },
        "credentialSubject":
        {
            "id": "<subjectDid>"
        }
    }
}
```

vp-sample.json
```
{
    "vc_policies":
    [
        "not-before"
    ],
    "request_credentials":
    [
        "OpenBadgeCredential"
    ]
}
```

### PR DESCRIPTION
**Pull request by mikeplotean**
## Description

Adds `validFrom` handling for `NotBeforeDatePolicy`. Adds tests for `NotBeforeDatePolicy` and `ExpirationDatePolicy`. Adds `DatePolicyUtils` for shared functionality.

Fixes #559 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] ~~Breaking change (fix or feature that would cause existing functionality to not work as expected)~~
- [ ] ~~This change requires a documentation update~~

## Checklist

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests
- [ ] ~~Do we need to implement analytics?~~
- [ ] ~~Will this be part of a product update? If yes, please write one phrase about this update.~~
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: waltid-libraries/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/verification/DatePolicyUtils.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        claims.firstOrNull { claim ->
+            !data?.jsonObject?.get(claim.getValue())?.jsonPrimitive?.content.isNullOrEmpty()
+        }?.let {
+            Pair(it, converter(data!!.jsonObject[it.getValue()]!!.jsonPrimitive.content))
+        }
+
+    private fun instantConverter(value: String) = Instant.parse(value)
+
+    private fun epochInstantConverter(value: String) =
+        value.toLongOrNull()?.let { Instant.fromEpochSeconds(it) } ?: error("Couldn't parse epoch seconds from: $value")
```
Comment (SuperBatata) [inline waltid-libraries/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/verification/DatePolicyUtils.kt:32]: Maybe using a more specific Error Handling option is better

**Review by SuperBatata — CHANGES_REQUESTED** [review]

File: waltid-libraries/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/verification/DatePolicyUtils.kt (hunk 1 above)
Comment (mikeplotean) [reply waltid-libraries/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/verification/DatePolicyUtils.kt:32]: do you mean throwing a different exception than `IllegalStateException`?

Comment (mikeplotean) [reply waltid-libraries/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/verification/DatePolicyUtils.kt]: do you think `IllegalArgumentException` is a better option? or maybe you mean smth. different?

Comment (SuperBatata) [reply waltid-libraries/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/verification/DatePolicyUtils.kt]: Yes `IllegalArgumentException` is a good fit for this

**Review by SuperBatata — APPROVED** [review]

**Pull request merged**